framework/Models/Code.py

```diff
@@ -308,8 +312,11 @@ def createNewInput(self,currentInput,samplerType,**kwargs):
     if not os.path.exists(subDirectory):
       os.mkdir(subDirectory)
     for index in range(len(newInputSet)):
-      newInputSet[index].setPath(subDirectory)
-      shutil.copy(self.oriInputFiles[index].getAbsFile(),subDirectory)
+      subSubDirectory = os.path.join(subDirectory,newInputSet[index].subDirectory)
+      if newInputSet[index].subDirectory.strip() != "" and not os.path.exists(subSubDirectory):
+        os.mkdir(subSubDirectory)
```

Comment:
We should determine if these lines of code are reachable currently. If not, then it may make more sense to leave them as commented out or for now put a comment above them that states they may be valuable pending future development. I can try to amend this and merge it as this was my only outstanding issue.